### ISSUE TITLE
Rc has wrong api code

### ISSUE THREAD

**Issue by eileenmcnaughton** [civicrm/org.civicrm.api4#198]
@colemanw @totten I think we need to tag a release to get the right api code for the rc

**Comment by totten** [comment]
OK, so we've generally agreed that we need to get api4 merged into core. Coleman has been hesitant to do a release (*in any form - whether by tagging ext or merging into core*) with the current contract for custom-data and n-to-1 joins. However, with #199, he's happier with the contract and on-board for a new release.

It's technically a breaking change, so we wanted some accommodation in case there's someone that needs the previous behavior. There's also some interaction with the core test suite that can be tricky.

We negotiated a path for releasing/merging the next APIv4 - I've put that in the description

**Comment by seamuslee001** [comment]
Added item about a test runner in civi-test-run see https://github.com/civicrm/civicrm-buildkit/pull/474/files

**Comment by colemanw** [comment]
I checked out the PR branch and then tried to disable the api4 extension but got `PHP Fatal error:  Cannot redeclare civicrm_api4()`. Whoops, this is going to be a bit of a chicken-n-egg problem when trying to upgrade sites that have the extension already. Have added task 17 to check/resolve upgrade issues.

**Comment by colemanw** [comment]
@seamuslee001 could you do number 11?

**Comment by seamuslee001** [comment]
so @colemanw always use 4.5 for 5.17 and 5.18 and 4.4 for anything before that?

**Comment by colemanw** [comment]
Yes, although I don't think our infra runs tests on anything before that.

**Comment by seamuslee001** [comment]
@colemanw i think we still run 5.13 tests as that is an ESR i think

**Comment by seamuslee001** [comment]
also does this mean @colemanw @totten we should remove api4 from the ext-matrix or keep there but just skip for master/5.19 and up

**Comment by seamuslee001** [comment]
@colemanw i note that this is showing 4.4.2 for 5.17 https://civicrm.org/extdir/ver=5.17%7Ccms=Drupal%7Cstatus=%7Cready=/org.civicrm.api4.xml is that correct?

**Comment by colemanw** [comment]
@seamuslee001 I would remove it from the ext matrix for the sake of simplicity. Also won't be needing the CI on the api4 PR queue anymore.

**Comment by colemanw** [comment]
> @colemanw i note that this is showing 4.4.2 for 5.17 https://civicrm.org/extdir/ver=5.17%7Ccms=Drupal%7Cstatus=%7Cready=/org.civicrm.api4.xml is that correct?

Yes that's correct @seamuslee001 we require 5.18 for 4.5.x

**Comment by totten** [comment]
Note that if we merged all this today, the core-bundled version would ship Nov 6.

I'd be in favor of removing the `api4` ext from the ext-matrix after a couple months. (*i.e. Wait until at least Nov 6 - maybe Dec 4 or Jan 1.*) There's no concrete expectation of doing another release of the ext... but, hypothetically, if there were some kind of critical/security issue in api4 ext in the next couple months, then we'd probably be on the hook for an ext release.
